File: algorithms/ar-bsrh/cpp/LinearSearch.cpp
#include <iostream>
using namespace std;

bool linearSearch(int arr[], int x, int n) {
    for(int i = 0; i < n; ++i) {
        if(arr[i] == x) {
            return true;
        }
    }
    return false;
}

int main() {
    int n, x, q;
    cout << "Enter size of array: ";
    cin >> n;
    int arr[n];
    cout << "Enter array elements space separated: ";
    for(int& i : arr) {
        cin >> i;
    }
    
    cout << "Enter number of queries you want: ";
    cin >> q;
    for(int i = 0; i < q; ++i) {
        cout << "Enter a number to search in array: ";
        cin >> x;
        if(linearSearch(arr, x, n)) {
            cout << x << " is present in the array.\n";
        }else {
            cout << x << " is not present in the array.\n";
        }
    }
    return 0;
}

File: algorithms/ar-bsrh/java/LinearSearch.java
import java.util.*;

public class LinearSearch {
    public static void main (String[] args) {
        Scanner sc = new Scanner(System.in);
        int n, x, q;
        System.out.print("Enter size of array: ");
        n = sc.nextInt();
        int arr[] = new int[n];
        System.out.print("Enter array elements space separated: ");
        for(int i = 0; i < n; ++i) {
            arr[i] = sc.nextInt();
        }
        
        System.out.print("Enter number of queries you want: ");
        q = sc.nextInt();
        for(int i = 0; i < q; ++i) {
            System.out.print("Enter a number to search in array: ");
            x = sc.nextInt();
            if(linearSearch(arr, x, n)) {
                System.out.print(x + " is present in the array.\n");
            }else {
                System.out.print(x + " is not present in the array.\n");
            }
        }
    }
    
	public static boolean linearSearch(int arr[], int x, int n) {
        for(int i = 0; i < n; ++i) {
            if(arr[i] == x) {
                return true;
            }
        }
        return false;
	}
}
